src/scheduler/node_info.c

```diff
@@ -4763,7 +4763,7 @@ reorder_nodes(node_info **nodes, resource_resv *resresv)
 	if (last_node_name[0] == '\0')
 		strcpy(last_node_name, nodes[0]->name);
 
-	if (resresv->aoename != NULL && conf.provision_policy == AVOID_PROVISION) {
+	if (resresv != NULL && resresv->aoename != NULL && conf.provision_policy == AVOID_PROVISION) {
```

Comment:
Wouldn't it be better to factor the if(resresv != NULL) out of this block and the block two higher and but both of them in an if (reseresv != NULL) if block?

Reply:
I could do that, I just wasn't comfortable changing a whole lot of code that I'm not working on and then end up with doing a lot of testing around it.
If you feel strongly about it then I'll change it, else I'd like to leave it as is.

Reply:
It's redundant though.  I'd rather do it right instead of having redundant checks.